network/src/protocols/wire/handshake/v1/mod.rs

```diff
@@ -12,23 +13,124 @@
 
 use serde::{Deserialize, Serialize};
 use serde_repr::{Deserialize_repr, Serialize_repr};
-use std::collections::HashMap;
+use std::{collections::BTreeMap, convert::TryInto, iter::Iterator};
 
 #[cfg(test)]
 mod test;
 
+/// Unique identifier associated with each application protocol.
+/// New application protocols can be added without bumping up the MessagingProtocolVersion.
+#[repr(u8)]
+#[derive(Clone, Copy, Debug, Hash, Eq, PartialEq, Deserialize_repr, Serialize_repr)]
```

Comment:
as you mentioned in the earlier comment, with @matbd's varint support in lcs, we could also remove the repr(u8) here?

Reply:
Yes, I wanted to do that too..however, since I want to do operations such as `ProtocolId::Consensus as u8` to set in my bitvec, I can't remove the `#[repr(u8)]` part. I can definitely get rid of `Serialize/Deserialize_repr` though, which is great. Thanks! :-)